### ISSUE TITLE
No way to get bug fixes + set version to 3.3 (so exclude 3.4 syntax)

### ISSUE THREAD

**Issue by eregon** [ruby/prism#2163]
From https://github.com/ruby/prism/pull/2118#discussion_r1445987020

@andrykonchin and I are looking at which version we'd set for truffleruby 24.0 (the next feature release where we'll switch to Prism).
We do want/need the flag from https://github.com/ruby/prism/pull/2118 to be set (as well as other fixes/improvements).
But with the current code in https://github.com/ruby/prism/pull/2118 that means we'd need to use PM_OPTIONS_VERSION_LATEST.

OTOH we do not want `it` handling from 3.4 (https://github.com/ruby/prism/pull/2124) for truffleruby (if at all possible).
And using `PM_OPTIONS_VERSION_LATEST` would enable `it` (when that PR is merged).

So how to solve this?
Maybe we need `PM_OPTIONS_VERSION_CRUBY_3_3` (without _0) to mean 3.3 syntax with the latest fixes?
Alternatively, is it worth hiding the flag from https://github.com/ruby/prism/pull/2118 for Prism versions newer than the one shipped with 3.3.0? Maybe not?
It feels like this flag should not hurt anything, and if someone wants exactly the version of Prism as shipped in 3.3.0 then they can just specify that Prism release in their Gemfile.
In short adding PM_CALL_NODE_FLAGS_ATTRIBUTE_WRITE is not really a syntax change or different parsing or anything, it's just extra information or from a POV a bug fix for a missing flag.

@kddnewton WDYT?

_Originally posted by @eregon in https://github.com/ruby/prism/pull/2118#discussion_r1445987020_

**Comment by eregon** [comment]
Thank you, that works, then we'll use `PM_OPTIONS_VERSION_CRUBY_3_3_0` in TruffleRuby to target Ruby 3.3.0 syntax.
